### PR TITLE
sqlccl: refuse to overwrite existing backup

### LINKED ISSUE #15051: sqlccl: `backup` should not overwrite an existing backup

### PR DESCRIPTION
**Pull request by dt**
Fixes #15051

### CODE REVIEW

File: pkg/ccl/sqlccl/backup.go [hunk 1]
```diff
@@ -271,10 +271,24 @@ func Backup(
...
+	exportStore, err := exportStorageFromURI(ctx, uri)
 	if err != nil {
 		return BackupDescriptor{}, err
 	}
+	defer exportStore.Close()
+
+	// Ensure there isn't already a readable backup desc.
+	{
+		r, err := exportStore.ReadFile(ctx, BackupDescriptorName)
+		// TODO(dt): If audit/wrap the not-exists errors of all ExportStorage
```
Comment (mjibson) [inline pkg/ccl/sqlccl/backup.go]: I don't understand this comment. Can you rephrase it?

**Comment by tamird** [comment]
:lgtm: modulo the broken comment.

---

Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056#-:-Ki1fSJqeAqoTkpultcF:befv9ps)*

**Comment by dt** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[pkg/ccl/sqlccl/backup.go, line 283 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056#-Ki1YMMLOklJcQ-sQtM9-r1-283:-Ki1fU_kjY4njdCfJp8C:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6010456578323fe27880c564b79adb4dceb11dc6/pkg/ccl/sqlccl/backup.go#L283)):*
<details><summary><i>Previously, mjibson (Matt Jibson) wrote…</i></summary><blockquote>

I don't understand this comment. Can you rephrase it?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056)*

**Comment by danhhz** [comment]
:lgtm: ditto comment

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056#-:-Ki1fap9msYkihnPkTu3:bkm5tve)*

**Comment by dt** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/ccl/sqlccl/backup.go, line 283 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056#-Ki1YMMLOklJcQ-sQtM9-r1-283:-Ki1fzDWSa8uvmqItNVq:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/6010456578323fe27880c564b79adb4dceb11dc6/pkg/ccl/sqlccl/backup.go#L283)):*
<details><summary><i>Previously, dt (David Taylor) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056)*

**Comment by tamird** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/15056)*

**Review by madelynnblue — APPROVED** [review]
